### ISSUE TITLE
Improve scrollbars

### ISSUE THREAD

**Issue by nicolas-p** [factor/factor#1579]
I was going to open an issue about scrollbars, and @bpollack said yesterday that he wanted to fix them.

I improved them slightly last summer by making the background images transparent (or white ?) but I didn't touch the code.

- I think scroller gadgets are organized in four panes: content, vertical scrollbar, horizontal scrollbar, and a remaining white square. Scrollbars always take space even if they're not needed. Factor already manages to hide the scroller if it's not needed, it should be possible to remove the space that it takes.

- Maybe we could remove the arrows used to move the scrollbars. Does anybody use them?

- Scrolling is not smooth at all, it moves by five or ten pixels at a time. This might not be easy to fix.

**Comment by mrjbq7** [comment]
Drawing them on top of the content like Chrome or Safari would be neat.

**Comment by nicolas-p** [comment]
I have found a way to make scrollers disappear completely when not needed, but it could probably be coded better.

```factor
IN: ui.gadgets.scrollers
M: scroller layout*
    dup
    [ x>> dup thumb>> visible?>> [ { 100 16 } ] [ { 0 0 } ] if >>pref-dim drop ] 
    [ y>> dup thumb>> visible?>> [ { 16 100 } ] [ { 0 0 } ] if >>pref-dim drop ] bi
    [ call-next-method ] [
        dup follows>>
        [ update-scroller ] [ >>follows drop ] 2bi
    ] bi ;
```

**Comment by mrjbq7** [comment]
Does the content correctly take up more space?

I was looking at this awhile ago and thought trying to emulate the drawn over top scrollbar might be a better but slightly tricker approach?

**Comment by nicolas-p** [comment]
Yes the content takes all the space when there are no scrollbars. The only problem is that when you resize the window, the "thumb" appears one iteration before space is taken by the slider, and it disappears one iteration before space is given back to the content. I have tried to do it differently but I always have the same problem.

**Comment by mrjbq7** [comment]
Fixed by #1618.
